gatsby-config.js

```diff
@@ -5,16 +5,26 @@ const activeEnv =
 
 console.log(`Using environment config: '${activeEnv}'`);
 
-const siteUrl =
-  process.env.GATSBY_SITE_URL ||
-  (process.env.CONTEXT === "production"
-    ? process.env.URL
-    : process.env.DEPLOY_PRIME_URL);
-
 require("dotenv").config({
   path: `.env.${activeEnv}`,
 });
 
+// Support for Gatsby CLI
+let siteUrl = 'http://localhost:8000';
+
+// Support for Production site builds.
+if (process.env.CONTEXT === 'production') {
+  siteUrl = process.env.URL;
+}
+// Support for non-production netlify builds (branch/preview)
+else if (process.env.CONTEXT !== 'production' && process.env.NETLIFY) {
+  siteUrl = process.env.DEPLOY_PRIME_URL;
+}
+// Support for Netlify CLI.
+else if (process.env.NETLIFY_DEV === true) {
+  siteUrl = 'http://localhost:64946';
+}
+
```

Comment:
@sherakama @mattanglin I was reviewing the image focal point PR, and found that there's a graphQL error in the useSiteMetadata hook, so I borrowed this bit of code from the saa_alumni repo which seems to solve the problem. Could you please check that I'm not adding anything extra here? I suspect we might not need the last `else if` for this site actually?

Found another issue below.....

Reply:
That all looks good to me? The last `else if` shouldn't harm anything. It handles the `siteUrl` when running `netlify dev` locally. It is curious that the `siteUrl` was causing GraphQL errors, though. Typically that only happens when a field is `undefined` or `null` so GraphQL drops it from its queryable interface; i.e. That error signifies to me that `siteUrl` isn't getting set for the build.

Reply:
Yeah it caused an error when running locally (npm run develop) but it was ok for the Netlify preview 🤔

Reply:
Looking at the old code, I imagine it's because there was no fallback `siteUrl` set for local dev? It would probably be fine if you set `GATSBY_SITE_URL=http://localhost:8000` in your local `.env`, but this updated `siteUrl` resolution is more complete and explicit anyhow.

Reply:
Yeah, that is probably why. It's been a while since I run this repo locally 😃